### PR TITLE
net-wireless/unifi: New package

### PR DESCRIPTION
**Pull request by ConiKost**
net-wireless/unifi: New package

v5.6.39 = LTS
v5.7.28 = Current Stable
v5.8.23 = Next Stable
v5.9.4 = Unstable (dropped keywords)

Closes: https://bugs.gentoo.org/634518

### CODE REVIEW

File: net-wireless/unifi/unifi-5.7.23.ebuild [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit systemd user
+
+DESCRIPTION="A Management Controller for Ubiquiti Networks UniFi APs"
+HOMEPAGE="https://www.ubnt.com"
+SRC_URI="https://dl.ubnt.com/unifi/${PV}/UniFi.unix.zip -> ${P}.zip"
```
Comment (Zlogene) [inline net-wireless/unifi/unifi-5.7.23.ebuild]: add `app-arch/unzip` to `DEPEND`

File: net-wireless/unifi/unifi-5.7.25.ebuild [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+EAPI=6
+
+MY_PV="8d34344384"
+
+inherit systemd user
+
+DESCRIPTION="A Management Controller for Ubiquiti Networks UniFi APs"
+HOMEPAGE="https://www.ubnt.com"
+SRC_URI="https://dl.ubnt.com/unifi/${PV}-${MY_PV}/UniFi.unix.zip -> ${P}.zip"
```
Comment (Zlogene) [inline net-wireless/unifi/unifi-5.7.25.ebuild]: add `app-arch/unzip` to `DEPEND`

File: net-wireless/unifi/unifi-5.7.23.ebuild (hunk 1 above)
Comment (ConiKost) [reply net-wireless/unifi/unifi-5.7.23.ebuild]: Fixed.

File: net-wireless/unifi/unifi-5.7.25.ebuild (hunk 2 above)
Comment (ConiKost) [reply net-wireless/unifi/unifi-5.7.25.ebuild]: Fixed.

File: profiles/license_groups [hunk 3]
```diff
@@ -71,7 +71,7 @@ BINARY-REDISTRIBUTABLE @FREE bh-luxi Broadcom Dina intel-ucode ipw2100-fw ipw220
```
Comment (mgorny) [inline profiles/license_groups]: Did you mean: 'grue'?

File: net-wireless/unifi/files/unifi.initd [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+UNIFI_PATH="/var/lib/unifi"
+UNIFI_USER="unifi"
+
+depend() {
+	need net
+}
+
+start() {
+	ebegin "Starting UniFi Controller"
+	cd "${UNIFI_PATH}"
```
Comment (mgorny) [inline net-wireless/unifi/files/unifi.initd]: `&&`, or this will fail silently.

File: net-wireless/unifi/unifi-5.6.37.ebuild [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+DEPEND="app-arch/unzip"
+
+RESTRICT="bindist mirror strip"
+
+S="${WORKDIR}/UniFi"
+
+DOCS=( "readme.txt" )
+
+QA_PREBUILT="usr/lib/unifi/lib/native/Linux/x86_64/*.so
```
Comment (mgorny) [inline net-wireless/unifi/unifi-5.6.37.ebuild]: Do they install a directory called `x86_64` on 32-bit systems?

File: net-wireless/unifi/unifi-5.6.37.ebuild [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+	fi
+
+	insinto /usr/$(get_libdir)/unifi/lib
+	doins lib/*.jar
+
+	exeinto /usr/$(get_libdir)/unifi/lib/native/Linux/x86_64
+	doexe lib/native/Linux/x86_64/*.so
+
+	for dir in conf run tmp work; do
+		keepdir /var/lib/unifi/${dir}
```
Comment (mgorny) [inline net-wireless/unifi/unifi-5.6.37.ebuild]: `keepdir /var/lib/unifi/{conf,run,tmp,work}`?

File: net-wireless/unifi/unifi-5.6.37.ebuild [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+	done
+	keepdir /var/lib/unifi
```
Comment (mgorny) [inline net-wireless/unifi/unifi-5.6.37.ebuild]: Why? You've already kept all four subdirectories.

File: net-wireless/unifi/unifi-5.6.37.ebuild [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+	dosym ../../../var/log/unifi /usr/$(get_libdir)/unifi/logs
+	dosym ../../../usr/$(get_libdir)/unifi/lib /var/lib/unifi/lib
+
+	newinitd "${FILESDIR}"/${PN}.initd unifi
+	systemd_dounit "${FILESDIR}"/${PN}.service
+
+	echo 'CONFIG_PROTECT="/var/lib/unifi"' > "${T}"/99unifi || die
+	doenvd "${T}"/99unifi
+
+	fowners -R unifi:unifi /usr/$(get_libdir)/unifi /var/lib/unifi /var/log/unifi
```
Comment (mgorny) [inline net-wireless/unifi/unifi-5.6.37.ebuild]: This looks undesirable. Why would you let the software write to `/usr`?

File: profiles/license_groups (hunk 3 above)
Comment (ConiKost) [reply profiles/license_groups]: Sorry, what do you mean?

Comment (mgorny) [reply profiles/license_groups]: Read your commit message.

Comment (ConiKost) [reply profiles/license_groups]: Ah, I see, thanks!

File: net-wireless/unifi/unifi-5.6.37.ebuild (hunk 5 above)
Comment (ConiKost) [reply net-wireless/unifi/unifi-5.6.37.ebuild]: Yes. There is no `i386`. I just ran a test on Ubuntu 16.04 32-Bit. It also there install this folder.

File: profiles/license_groups (hunk 3 above)
Comment (ConiKost) [reply profiles/license_groups]: Fixed.

File: net-wireless/unifi/files/unifi.initd (hunk 4 above)
Comment (ConiKost) [reply net-wireless/unifi/files/unifi.initd]: Thanks, fixed.

File: net-wireless/unifi/unifi-5.6.37.ebuild (hunk 6 above)
Comment (ConiKost) [reply net-wireless/unifi/unifi-5.6.37.ebuild]: Thanks, changed.

File: net-wireless/unifi/unifi-5.6.37.ebuild (hunk 7 above)
Comment (ConiKost) [reply net-wireless/unifi/unifi-5.6.37.ebuild]: Yep, you are right, removed.

File: net-wireless/unifi/unifi-5.6.37.ebuild (hunk 8 above)
Comment (ConiKost) [reply net-wireless/unifi/unifi-5.6.37.ebuild]: Yes, I changed this now, as no write access is needed for `/usr/lib/unifi`.

**Comment by mgorny** [comment]
@gentoo/java, could you take a final look?

**Comment by ConiKost** [comment]
Rebased!

**Comment by chewi** [comment]
Tomcat-based stuff is tricky to do properly. Be aware that although you've depended on dev-java/tomcat-native, it will not use the jar from this, just the native library. No doubt the jar is bundled, along with the rest of Tomcat. Sucks but hey. Also be aware that although you've depended on virtual/jre:1.8, there's no guarantee that this will be the Java version that /usr/bin/java points to when you run it. It is at least highly likely for the time being because Java 9+ as a system VM is masked.

**Comment by ConiKost** [comment]
>  Be aware that although you've depended on dev-java/tomcat-native, it will not use the jar from this, just the native library.

Sorry, I don't understand, what you exactly mean?

> No doubt the jar is bundled, along with the rest of Tomcat. Sucks but hey.

You mean `tomcat libs`? Yes, I am aware of it.

> Also be aware that although you've depended on virtual/jre:1.8, there's no guarantee that this will be the Java version that /usr/bin/java points to when you run it.

Good catch. Is there anything I can do about? Or at least some `ewarn` at the end?

> It is at least highly likely for the time being because Java 9+ as a system VM is masked.

Yes, I hope, Ubiquiti Networks will start supporting soon Java 9+ in future, as the support officially Ubuntu and Ubuntu will drop Java 8 in future. Currently, there is a red warning `As of 5.7.x we only support Java 8. At this time Java 9 is not supported.` I haven't yet tested, if it, regarding the warning, it could work with Java 9+.

**Comment by chewi** [comment]
@benkohler, why did you merge this when we were still discussing it? I know I sometimes hold these things up but not all of us have so much time.

**Comment by ConiKost** [comment]
> @benkohler, why did you merge this when we were still discussing it?

It seems even incomplete, as not all versions got merged? And an older version, not my current commits?!

**Comment by benkohler** [comment]
I've been working with ConiKost to get my changes merged with his, and the current PR is the result of his work on that.  Am I ok to merge these changes?

**Comment by benkohler** [comment]
Pushed

**Pull request closed without merging**